pkg/network/node/networkpolicy.go

```diff
@@ -303,7 +304,7 @@ func (np *networkPolicyPlugin) generateNamespaceFlows(otx ovs.Transaction, npns
 			for _, flow := range npp.flows {
 				otx.AddFlow("table=80, priority=150, reg1=%d, %s actions=output:NXM_NX_REG2[]", npns.vnid, flow)
 			}
-			if npp.selectedIPs == nil {
+			if npp.selectedIPs == nil || npp.matchesAllIPs {
 				allPodsSelected = true
```

Comment:
You don't want `||` here; the idea is that we're changing it so that `npp.selectedIPs == nil` no longer means "select all IPs". It means "select _no_ IPs". So you'd want just

    if npp.matchesAllIPs {
            allPodsSelected = true
    }

Reply:
Done.